File: Brute Force/BFS.cpp
#include<iostream>
#include<list>
using namespace std;

class graph
{
    int v;
    list <int> *adj;
public:
    graph(int v)
    {
        this->v=v;
        adj=new list <int>[v];
    }
    void addedge(int v,int w)
    {
        adj[v].push_back(w);
    }
    void bfs(int s)
    {
        bool *visited=new bool [v];
        for(int i=0;i,v;i++)
        {
            visited[i]=false;
        }

        list <int> queue;
        visited[s]=true;
        queue.push_back(s);
        list <int> :: iterator i;

        while(!queue.empty())
        {
            s=queue.front();
            cout<<s<<" ";
            queue.pop_front();

            for(i=adj[s].begin();i!=adj[s].end();++i)
            {
                if(!visited[*i])
                {
                    visited[*i]=true;
                    queue.push_back(*i);
                }
            }
        }

    }
};

int main()
{
    graph g(5);

    g.addedge(1,2);
    g.addedge(1,3);
    g.addedge(1,5);
    g.addedge(2,3);
    g.addedge(2,4);
    g.addedge(4,5);
    g.addedge(4,3);

    g.bfs(1);

}

File: Graphs/JavaScript/bfs.js
function bfs(graph,visited,n,sv){
    let queue = [];
    queue.push(sv);
    visited[sv] = true;
    while(queue.length >= 1){
        let element = queue.shift();
       // console.log('element',element)
        console.log(element);
        for(let i=0; i<n; i++){
            if(i === element) continue;
            if(graph[element][i] === 1 && !visited[i]){
                queue.push(i);
                visited[i] = true;
            }
        }
    }
}

let graph = [
    [0,1,0,0,0,0,0,0],
    [1,0,1,0,0,0,0,0],
    [0,1,0,1,1,0,0,0],
    [0,0,1,0,0,0,1,0],
    [0,0,1,0,0,1,0,0],
    [0,0,0,0,1,0,0,1],
    [0,0,0,1,0,0,0,1],
    [0,0,0,0,0,1,1,0],
]

let visited = [false,false,false,false,false,false,false,false]

bfs(graph,visited,8,0);